File: frontend/src/types/supplier.ts
export interface Supplier {
  id: string;
  invite_token: string;
  email: string;
  email_verified: boolean;
  company_name: string | null;
  company_address: string | null;
  tax_id: string | null;
  npi: string | null;
  primary_contact_name: string | null;
  primary_contact_title: string | null;
  primary_contact_email: string | null;
  primary_contact_phone: string | null;
  escalation_contact_name: string | null;
  escalation_contact_email: string | null;
  escalation_contact_phone: string | null;
  tier: "tier_1" | "tier_2" | null;
  order_transmittal_preference: "secure_email" | "fax" | "api" | null;
  transmittal_destination: string | null;
  shipping_fee_structure: ShippingFeeStructure | null;
  return_policy: ReturnPolicy | null;
  support_hours: string | null;
  support_phone: string | null;
  support_email: string | null;
  after_hours_process: string | null;
  stripe_account_id: string | null;
  stripe_onboarding_complete: boolean;
  sla_acknowledged: boolean;
  sla_acknowledged_by: string | null;
  sla_acknowledged_at: string | null;
  current_step: number;
  status: SupplierStatus;
  submitted_at: string | null;
  approved_at: string | null;
  created_at: string;
  updated_at: string;
}

export type SupplierStatus =
  | "in_progress"
  | "submitted"
  | "under_review"
  | "action_needed"
  | "approved"
  | "live";

export interface ShippingFeeStructure {
  free_shipping_threshold?: number;
  flat_rate?: number;
  is_variable?: boolean;
  fee_schedule?: string;
}

export interface ReturnPolicy {
  return_window_days?: number;
  restocking_fee?: number;
  restocking_fee_type?: "percentage" | "flat";
  return_shipping_responsibility?: "supplier" | "patient";
  condition_requirements?: string[];
}

export interface Product {
  id: string;
  supplier_id: string;
  product_name: string;
  hcpcs_code: string | null;
  category: string | null;
  retail_price: number | null;
  hcpcs_fee_schedule: Record<string, unknown> | null;
  sku: string | null;
  manufacturer: string | null;
  variant_size: string | null;
  fulfillment_types: string[] | null;
  ai_confidence: Record<string, string> | null;
  approved_by_supplier: boolean;
  created_at: string;
  updated_at: string;
}

export interface Payer {
  id: string;
  supplier_id: string;
  payer_name: string;
  network_type: string;
  created_at: string;
}

export interface PayerExclusion {
  id: string;
  supplier_id: string;
  payer_id: string;
  product_id: string | null;
  category: string | null;
  created_at: string;
}

export interface ServiceArea {
  id: string;
  supplier_id: string;
  state: string;
  cities: string[] | null;
  zip_codes: string[] | null;
  standard_delivery_days: number | null;
  expedited_delivery_days: number | null;
  created_at: string;
  updated_at: string;
}

export interface CatalogUpload {
  upload_id: string;
  original_filename: string;
  processing_status: "uploaded" | "processing" | "completed" | "failed";
}

export interface StepConfig {
  number: number;
  label: string;
  key: string;
  requiresTier1?: boolean;
}

export const ONBOARDING_STEPS: StepConfig[] = [
  { number: 1, label: "Company Information", key: "company-info" },
  { number: 2, label: "Tier Selection", key: "tier-selection" },
  { number: 3, label: "Product Catalog", key: "product-catalog" },
  { number: 4, label: "Accepted Payers", key: "accepted-payers" },
  { number: 5, label: "Payer Exclusions", key: "payer-exclusions" },
  { number: 6, label: "Service Areas", key: "service-areas" },
  { number: 7, label: "Operations Setup", key: "operations-setup" },
  { number: 8, label: "Order Transmittal", key: "order-transmittal" },
  { number: 9, label: "Stripe Setup", key: "stripe-setup", requiresTier1: true },
  { number: 10, label: "SLAs & Agreements", key: "sla" },
  { number: 11, label: "Review & Submit", key: "review-submit" },
];

File: backend/app/schemas/supplier.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, EmailStr


class SupplierRead(BaseModel):
    id: UUID
    invite_token: str
    email: str
    email_verified: bool
    company_name: Optional[str] = None
    company_address: Optional[str] = None
    tax_id: Optional[str] = None
    npi: Optional[str] = None
    operations_contact_name: Optional[str] = None
    operations_contact_title: Optional[str] = None
    operations_contact_email: Optional[str] = None
    operations_contact_phone: Optional[str] = None
    escalation_contact_name: Optional[str] = None
    escalation_contact_title: Optional[str] = None
    escalation_contact_email: Optional[str] = None
    escalation_contact_phone: Optional[str] = None
    tier: Optional[str] = None
    order_transmittal_preference: Optional[str] = None
    transmittal_destination: Optional[str] = None
    shipping_fee_structure: Optional[dict] = None
    return_policy: Optional[dict] = None
    support_hours: Optional[str] = None
    support_phone: Optional[str] = None
    support_email: Optional[str] = None
    after_hours_process: Optional[str] = None
    stripe_account_id: Optional[str] = None
    stripe_onboarding_complete: bool
    sla_acknowledged: bool
    sla_acknowledged_by: Optional[str] = None
    sla_acknowledged_at: Optional[datetime] = None
    current_step: int
    status: str
    submitted_at: Optional[datetime] = None
    approved_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SupplierUpdate(BaseModel):
    company_name: Optional[str] = None
    company_address: Optional[str] = None
    tax_id: Optional[str] = None
    npi: Optional[str] = None
    operations_contact_name: Optional[str] = None
    operations_contact_title: Optional[str] = None
    operations_contact_email: Optional[str] = None
    operations_contact_phone: Optional[str] = None
    escalation_contact_name: Optional[str] = None
    escalation_contact_title: Optional[str] = None
    escalation_contact_email: Optional[str] = None
    escalation_contact_phone: Optional[str] = None
    tier: Optional[str] = None
    order_transmittal_preference: Optional[str] = None
    transmittal_destination: Optional[str] = None
    shipping_fee_structure: Optional[dict] = None
    return_policy: Optional[dict] = None
    support_hours: Optional[str] = None
    support_phone: Optional[str] = None
    support_email: Optional[str] = None
    after_hours_process: Optional[str] = None
    sla_acknowledged: Optional[bool] = None
    sla_acknowledged_by: Optional[str] = None
    current_step: Optional[int] = None


class SupplierSubmitResponse(BaseModel):
    id: UUID
    status: str
    submitted_at: datetime


class SupplierStatusResponse(BaseModel):
    id: UUID
    status: str
    current_step: int
    submitted_at: Optional[datetime] = None
    approved_at: Optional[datetime] = None
